src/path.jl

```diff
@@ -20,18 +20,36 @@ Path(fp::AbstractPath) = fp
 
 # May want to support using the registry for other constructors as well
 function Path(str::AbstractString; debug=false)
-    types = filter(t -> ispathtype(t, str), PATH_TYPES)
+    result = nothing
+    types = Vector{eltype(PATH_TYPES)}()
+
+    for P in PATH_TYPES
+        r = tryparse(P, str)
+
+        # If we successfully parsed the path then save that result
+        # and break if we aren't in debug mode, otherwise record how many
+        if r !== nothing
+            result = r
+            if debug
+                push!(types, P)
+            else
+                break
+            end
+        end
+    end
 
     if length(types) > 1
         @debug(
             string(
-                "Found multiple path types that match the string specified ($types). ",
-                "Please use a specific constructor if $(first(types)) is not the correct type."
+                "Found multiple path types that could parse the string specified ($types). ",
+                "Please use a specific `parse` method if $(first(types)) is not the correct type."
             )
         )
+    elseif result == nothing
+        throw(ArgumentError("Unable to parse $str as a path type."))
+    else
+        return result
```

Comment:
should this be `tryparse{::Type{AbstractPath}, str::String)`

Reply:
Argument against that is that such a `tryparse` should maybe be checking *all* subtypes of `AbstractPath` not just the registered ones?

Reply:
Yeah, currently we provide a `tryparse{::Type{P}, str::String) where {P <: AbstractPath` method, but that's intended as a fallback to deprecate the old `ispathtype` approach. If we made the entire thing one big `tryparse` for `AbstractPath` then I'm not sure how we'd handle that deprecation?

Reply:
Deprecation is not required by SemVer.
Document it as breaking change in the release notes, and move on.

Reply:
👌